core/service/src/chain_ops.rs

```diff
@@ -155,6 +175,13 @@ pub fn import_blocks<F, E, R>(
 			info!("#{}", b);
 		}
 	}
+
+	let mut blocks_imported = 0;
+	while blocks_imported < count {
+		wait_recv.recv().expect("Unable to receive info about imported block");
+		blocks_imported += 1;
+	}
```

Comment:
As an idea: you could create a `WaitLink` and then count imported blocks there and send a signal only when all blocks were imported.

Not blocking on this though.

Reply:
Good idea, the `block_imported` function of the `Link` trait doesn't take a mutable `self` though. So I think this would require a more extensive refactoring.